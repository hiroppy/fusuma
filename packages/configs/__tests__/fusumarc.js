const fs = require('fs');
const path = require('path');
const fusuma = require('../src/fusumarc');

describe('fusuma', () => {
  const p = path.join(__dirname, '.fusumarc.yml');

  afterAll(() => {
    fs.unlinkSync(p);
  });

  test.skip('init', async () => {
    await fusuma.init(__dirname);

    expect(fs.existsSync(p)).toBeTruthy();
    expect(fs.readFileSync(p, 'utf8')).toMatchSnapshot();
  });

  test('read', async () => {
    // priority: yml > js

    fs.writeFileSync(
      path.join(__dirname, '.fusumarc.yml'),
      `meta:
  url:
  title: yml
  description:
  thumbnail:
  siteName:
slide:
  theme: js
extends:
  js:
  css:
    `
    );

    expect(await fusuma.read(__dirname)).toMatchSnapshot();

    fs.unlinkSync(p);

    fs.writeFileSync(
      path.join(__dirname, '.fusumarc.js'),
      `module.exports = {
  meta: {
    url: '',
    title: 'js',
    description: '',
    thumbnail: '',
    siteName: ''
  },
  slide: {
    theme: ''
  },
  extends: {
    js: '',
    css: ''
  }
};
    `
    );

    expect(await fusuma.read(__dirname)).toMatchSnapshot();

    fs.unlinkSync(path.join(__dirname, '.fusumarc.js'));
  });
});
