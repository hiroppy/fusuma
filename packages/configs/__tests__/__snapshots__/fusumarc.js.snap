// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fusuma init 1`] = `
"meta:
  url:
  title:
  thumbnail:
  description:
  sns:
    - twitter
slide:
  loop: true
  sidebar: true
  targetBlank: true
  showIndex: false
  isVertical: false
  code:
    languages:
      - javascript
    plugins: []
    theme: default
  chart: false
extends:
  js:
  css:
"
`;

exports[`fusuma read 1`] = `
Object {
  "extends": Object {
    "css": null,
    "js": null,
  },
  "meta": Object {
    "description": null,
    "siteName": null,
    "thumbnail": null,
    "title": "yml",
    "url": null,
  },
  "slide": Object {
    "theme": "js",
  },
}
`;

exports[`fusuma read 2`] = `
Object {
  "extends": Object {
    "css": "",
    "js": "",
  },
  "meta": Object {
    "description": "",
    "siteName": "",
    "thumbnail": "",
    "title": "js",
    "url": "",
  },
  "slide": Object {
    "theme": "",
  },
}
`;
